File: src/BreakfastProvider.Api/Models/Requests/PancakeRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record PancakeRequest
{
    public string? Milk { get; init; }

    public string? Flour { get; init; }

    public string? Eggs { get; init; }

    public List<string> Toppings { get; init; } = [];
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/PancakeRequest.java
package io.lemonlion.breakfast.model.request;

import java.util.ArrayList;
import java.util.List;

/** Twin of C# {@code PancakeRequest}. */
public record PancakeRequest(String milk, String flour, String eggs, List<String> toppings) {

    public PancakeRequest {
        if (toppings == null) {
            toppings = new ArrayList<>();
        }
    }
}
